src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -741,10 +937,14 @@ public void ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseN
             {
                 try
                 {
-                    if (info.SqlConnection.State == ConnectionState.Open)
+                    foreach (DbConnection connection in info.GetAllConnections())
                     {
-                        info.SqlConnection.ChangeDatabase(newDatabaseName);
+                        if (connection.State == ConnectionState.Open)
```

Comment:
I'm not totally sure about this, but can a connection go into a non-open state by being idle? If so, would we still need to change database anyways?